crates/rpc/eth/account.rs

```diff
@@ -88,6 +97,12 @@ pub fn get_balance(request: &GetBalanceRequest, storage: Store) -> Result<Value,
         "Requested balance of account {} at block {}",
         request.address, request.block
     );
+
+    // TODO: implement historical querying
+    if request.block != BlockTag::Latest {
+        return Err(RpcErr::Internal);
+    }
```

Comment:
This means that the case in which the block in the request is the hash of the latest case will also fail when it is a case that we can handle

Reply:
We could instead use `resolve_block_number` to get the block number and then check that it matches the last block number